consensus/src/experimental/commit_phase.rs

```diff
@@ -133,191 +135,211 @@ impl CommitPhase {
     pub fn new(
         commit_channel_recv: Receiver<(Vec<ExecutedBlock>, LedgerInfoWithSignatures)>,
         execution_proxy: Arc<dyn StateComputer>,
-        commit_msg_sender: channel::Sender<CommitPhaseMessage>,
         commit_msg_receiver: channel::Receiver<ConsensusMsg>,
         verifier: ValidatorVerifier,
         safety_rules: Arc<Mutex<MetricsSafetyRules>>,
         author: Author,
+        back_pressure: Arc<AtomicU64>,
+        network_sender: NetworkSender,
     ) -> Self {
         Self {
             commit_channel_recv,
             execution_proxy,
-            local_cache: HashMap::<HashValue, LedgerInfoWithSignatures>::new(),
-            local_queue: VecDeque::<PendingBlocks>::new(),
-            commit_msg_sender,
+            blocks: None,
             commit_msg_receiver,
             verifier,
             safety_rules,
             author,
-            committed_round: 0,
+            back_pressure,
+            network_sender,
         }
     }
 
     /// Notified when receiving a commit vote message
-    pub async fn process_commit_vote(&mut self, commit_vote: &CommitVote) -> anyhow::Result<()> {
-        let li = commit_vote.ledger_info();
-
-        if li.commit_info().round() < self.committed_round {
-            return Ok(()); // we ignore the message
-        }
+    pub async fn process_commit_vote(
+        &mut self,
+        commit_vote: &CommitVote,
+    ) -> anyhow::Result<(), Error> {
+        if let Some(pending_blocks) = self.blocks.as_mut() {
```

Comment:
we should also print something when the blocks is none to indicate why the message is ignored

Reply:
resolved